src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -573,73 +661,210 @@ void ompl::geometric::RRTstar::getPlannerData(base::PlannerData &data) const
     }
 }
 
-int ompl::geometric::RRTstar::pruneTree(const base::Cost pruneTreeCost)
+int ompl::geometric::RRTstar::pruneTree(const base::Cost& pruneTreeCost)
```

Comment:
This is a more 'admissible' pruning scheme.

What it does is iterate through the graph and flag every "leaf vertex" that meets the pruning condition (i.e., cannot provide a better solution). It then deletes up the branches from these leaves for as long as the vertices are (a) cannot provide a better solution and (b) a leaf.